### ISSUE TITLE
Sample code doesnt even work with v7

### ISSUE THREAD

**Issue by dsa0x** [Nerzal/gocloak#215]
```go
client := gocloak.NewClient("https://mycool.keycloak.instance")
	ctx := context.Background()
	token, err := client.LoginAdmin(ctx, "user", "password", "realmName")
	if err != nil {
		panic("Something wrong with the credentials or url")
	}
	user := gocloak.User{
		FirstName: "Bob",
		LastName:  "Uncle",
		Email:     "something@really.wrong",
		Enabled:   true,
		Username:  "CoolGuy",
	}
	_, err = client.CreateUser(ctx, token.AccessToken, "realm", user)
	if err != nil {
		panic("Oh no!, failed to create user :(")
	}
```

Sample code doesn't work in v7 because FirstName and the others want "*string" and not "string"

**Comment by SVilgelm** [comment]
So, this was long long discussion, it took a lot of iterations to came up with using the Pointers.
You need to pass the pointers instead of the direct values , unfortunately you cannot have the pointers of the constants in go, so you need to make the variables and pass them:
```go
	fName := "Bob"
	lName := "Uncle"
	email := "something@really.wrong"
	enabled := true
	uName := "CoolGuy"
	user := gocloak.User{
		FirstName: &fName,
		LastName:  &lName,
		Email:     &email,
		Enabled:   &enabled,
		Username:  &uName,,
	}
```

another option is you can use the special helpers from `utils.go`:
```go
	user := gocloak.User{
		FirstName: gocloak.StringP("Bob"),
		LastName:  gocloak.StringP("Uncle"),
		Email:     gocloak.StringP("something@really.wrong"),
		Enabled:   gocloak.BoolP(true),
		Username:  gocloak.StringP("CoolGuy"),
	}
```

**Comment by Nerzal** [comment]
We are open for better solutions to this problem, if there is any. 
@SVilgelm do you know which issue (nr) we discussed this?

**Comment by dsa0x** [comment]
> ```go
> 	fName := "Bob"
> 	lName := "Uncle"
> 	email := "something@really.wrong"
> 	enabled := true
> 	uName := "CoolGuy"
> 	user := gocloak.User{
> 		FirstName: &fName,
> 		LastName:  &lName,
> 		Email:     &email,
> 		Enabled:   &enabled,
> 		Username:  &uName,,
> 	}
> ```

Ohh. Thanks. I also had to do it this way. I just felt it should probably be changed in the readme as well

**Comment by timdrysdale** [comment]
Also discussed again in [Issue #237](https://github.com/Nerzal/gocloak/issues/237) (now closed) and fixed in [PR #240](https://github.com/Nerzal/gocloak/pull/240) so this can be closed now too.
